cupy/_statistics/order.py

```diff
@@ -323,9 +327,9 @@ def percentile(a, q, axis=None, out=None,
     if interpolation is not None:
         method = _check_interpolation_as_method(
             method, interpolation, 'percentile')
-    if not isinstance(q, cupy.ndarray):
-        q = cupy.asarray(q, dtype='d')
-    q = cupy.true_divide(q, 100)
+    if isinstance(q, (tuple, list)):
```

Comment:
I discussed offline about the change with @asi1024 
* This PR also improves the logic to avoid synchronization when `q` is given as `numpy.ndarray` or Python scalar (`list`/`tuple`)
* `float` is intentionally excluded here to compute the correct output dtype in `_quantile_unchecked`